scripts/link-checker

```diff
@@ -51,7 +51,7 @@ check() {
 	link=$(echo "$1" | grep -oE "(https|http|ftp):[^|]*")
 	files=$(echo "$1" | grep -oE "\|.*" | sed 's/|/ /g')
 
-	if grep -Fxq "$link" "$WHITELIST"; then
+	if grep -xq "$link" "$WHITELIST"; then
```

Comment:
This will (probably) not work. I know, because I already tried it 😊. One option to support regular expressions is to iterate through all patterns of the whitelist. My current solution for this approach looks like this:

```sh
while IFS= read -r pattern; do
	if printf '%s' "$link" | grep -q "$pattern"; then
		printf "whitelisted: “%s” because of pattern “%s”\n" "$link" "$pattern"
		printf "%i/%i\r" "$COUNTLINKS" "$NUMLINKS"
		return
	fi
done < "$WHITELIST"
```

Reply:
Thank you for pointing it out. I think I found a way around it, at least it worked locally.